### PR TITLE
Properly set options[:expires_in] from global_cache_options

### PR DESCRIPTION
**Pull request by fancyremarker**
This fixes the bug @hemslo caught in #91.

### CODE REVIEW

File: spec/garner/cache/identity_spec.rb [hunk 1]
```diff
@@ -7,6 +7,13 @@
     subject.options_hash[:foo].should eq 'bar'
   end
 
+  it 'includes Garner.config.global_cache_options[:expires_in]' do
+    Garner.configure do |config|
+      config.global_cache_options = { expires_in: 5.minutes }
+    end
```
Comment (dblock) [inline spec/garner/cache/identity_spec.rb]: This should be in a `before` block and you should clear this with a `Garner.reset!` or something similar in an `after` block.

**Comment by dblock** [comment]
See failing specs Mr. @fancyremarker :) LMK if you need help.

**Comment by dblock** [comment]
This also needs a CHANGELOG entry, please.

**Comment by fancyremarker** [comment]
@dblock: Roger that. Fixes coming. 2 points:

* The failure on spec/garner/mixins/mongoid/identity_spec.rb:48 is unrelated (it happens in master), but I'll try to fix here as well.
* The call to `Garner.config.reset!` happens in a [global before block](https://github.com/artsy/garner/blob/master/spec/support/garner.rb#L3).

**Comment by fancyremarker** [comment]
Rebased to upstream/master and fixed.

File: spec/garner/cache/identity_spec.rb [hunk 2]
```diff
@@ -1,14 +1,36 @@
...
+    it 'includes Garner.config.global_cache_options' do
+      expect(subject.options_hash[:foo]).to eq 'bar'
+    end
   end
 
-  it 'includes Garner.config.expires_in' do
-    Garner.configure { |config| config.expires_in = 5.minutes }
-    expect(subject.options_hash[:expires_in]).to eq 5.minutes
+  context 'with global_cache_options[:expires_in] defined ' do
+    before do
```
Comment (dblock) [inline spec/garner/cache/identity_spec.rb]: This `before` still needs a cleanup in `after`, otherwise you are polluting a potentially future test.

**Comment by dblock** [comment]
Squash your commits too, please.

File: spec/garner/cache/identity_spec.rb (hunk 2 above)
Comment (fancyremarker) [inline spec/garner/cache/identity_spec.rb]: See my comment. That already happens in a global `before`: https://github.com/artsy/garner/blob/master/spec/support/garner.rb#L3

**Comment by fancyremarker** [comment]
Squashed.

File: spec/garner/cache/identity_spec.rb (hunk 2 above)
Comment (dblock) [inline spec/garner/cache/identity_spec.rb]: :+1:

**Pull request merged**